.github/workflows/build.yml

```diff
@@ -31,18 +29,115 @@ jobs:
 
-      - uses: stefanzweifel/git-auto-commit-action@v4
+      - name: Upload build
+        uses: actions/upload-artifact@v3
         with:
-          commit_user_name: 'Wynntils'
-          commit_user_email: 'admin@wynntils.com'
-          commit_message: 'chore: spotless formatting'
+          name: build
+          path: |
+            forge/build/libs/wynntils-*.MC1.18.2-forge.jar
+            fabric/build/libs/wynntils-*.MC1.18.2-fabric.jar
+            quilt/build/libs/wynntils-*.MC1.18.2-quilt.jar
 
-      - name: Build with Gradle
-        uses: gradle/gradle-build-action@v2
+  changelog:
+    name: Generate Changelog
+    needs: [build]
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.changelog.outputs.tag }}
+      skipped: ${{ steps.changelog.outputs.skipped }}
+      clean_changelog: ${{ steps.changelog.outputs.clean_changelog }}
+      changelog: ${{ steps.changelog.outputs.changelog }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.PRIVATE_TOKEN }}
+
+      - name: Create changelog
+        id: changelog
+        uses: TriPSs/conventional-changelog-action@v3
         with:
-          arguments: build -x spotlessCheck --build-cache
+          github-token: ${{ secrets.PRIVATE_TOKEN }}
+          preset: conventionalcommits
+          release-count: 1
+          pre-commit: .pre-commit.js
+          create-summary: true
+          version-file: ./version.json
+          skip-git-pull: true
+
+  publish-gh:
+    name: Publish to Github Releases
+    runs-on: ubuntu-latest
+    needs: [ changelog, artifact ]
+    steps:
+      - uses: actions/download-artifact@v3
+
+      - name: Create release
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            - "forge/build/libs/wynntils-${{ steps.changelog.outputs.tag }}.MC1.18.2-forge.jar"
+            - "fabric/build/libs/wynntils-${{ steps.changelog.outputs.tag }}.MC1.18.2-fabric.jar"
+            - "quilt/build/libs/wynntils-${{ steps.changelog.outputs.tag }}.MC1.18.2-quilt.jar"
+          tag_name: ${{ needs.changelog.outputs.tag }}
+          prerelease: ${{ needs.changelog.outputs.pre_release }}
+          body: ${{ needs.changelog.outputs.clean_changelog }}
+
+  # Minecraft 1.18.2: 9008
+  # Forge: 7498
+  # Java 17: 8326
+  # Java 18: 8634
+  publish-forge:
+    name: Publish Forge to Curseforge & Modrinth
+    uses: Wynntils/Artemis/.github/workflows/publish.yml@main
+    needs: [build, changelog]
+    with:
+      file_path: "forge/build/libs/wynntils-${{ steps.changelog.outputs.tag }}.MC1.18.2-forge.jar"
+      release_name: "Wynntils (Artemis) ${{ steps.changelog.outputs.tag }}-forge"
+      tag: ${{ steps.changelog.outputs.tag }}
+      changelog: ${{ steps.changelog.outputs.changelog }}
+      cf_mc_version: "9008,7498,8326,8634"
+      mr_mc_version: 1.18.2
+      mr_loader: forge
+      secrets: inherit
+
+  # Minecraft 1.18.2: 9008
+  # Forge: 7499
```

Comment:
By this, do you mean `Fabric`? (Also same question about Quilt below)

Reply:
Yes 😅